content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,580 @@
+        cart.removeProductsFromCartByCustomer(1L, product);
+        assertEquals(2, cart.totalNumberOfItems());
+        cart=null;
+
+    }
+
+    @Test
+    void calculatingAllTheProductsInTheCart() {
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Customer customer=new Customer(1L, "Kingsley", "Chukwudi", new BigDecimal("2000"));
+        dataService.registerCustomer(customer);
+        cart.addProductToCartByCustomer(1L, product);
+        cart.addProductToCartByCustomer(1L, product2);
+        cart.addProductToCartByCustomer(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+
+        assertEquals(new BigDecimal( 3100), cart.calculatePrice());
+        customerDao=null;
+    }
+    @Test
+    void addProductToCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart=null;
+    }
+    @Test
+    void removeProductsFromCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        System.out.println(cart);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart.removeProductsFromCartByAdmin(1L,product);
+        System.out.println(cart);
+        assertEquals(2,cart.totalNumberOfItems());
+
+    }
+    @Test
+    void checkAllTheProductsInTheCartByCustomer(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Product product3 = new Product("3", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product3);
+        System.out.println(  cart.checkAllTheProductsInTheCartByCustomer(1L));
+        cart=null;
+    }
+    @Test
+    void checkAllRegisteredAdmin(){
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        System.out.println(dataService.getAllAdmin());
+    }
+}
```

Comment:
These tests aren't unit tests they are more so integration tests. Unit tests involve only a single class and test all the methods of that class (excluding getters and setters). Additionally, unit tests would use mocks of any dependency classes instead of the real instances and shouldn't start the Spring Context.